### PR TITLE
[stdlib] Make `DType` and `SIMD` work with `repr()`

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
I'll open issues for the TODO

I noticed that a `capacity` argument for String would be welcome and avoid having to create a list first.

### CODE REVIEW

File: stdlib/src/builtin/simd.mojo [hunk 1]
```diff
@@ -519,26 +520,65 @@ struct SIMD[type: DType, size: Int = simdwidthof[type]()](
...
+        Returns:
+            A string representation.
+        """
+        # TODO: Compute the exact amount of memory to store the representation.
+        var result = String(List[UInt8](capacity=32))
+        result += (
+            "SIMD[" + repr(self.element_type) + ", " + str(self.size) + "]("
+        )
+
+        # Usafe operations incoming, beware.
```
Comment (JoeLoser) [inline stdlib/src/builtin/simd.mojo:571]:
```suggestion
        # Unsafe operations incoming, beware.
```

**Review by JoeLoser — APPROVED** [review]
Nice!  Thanks for working on this.

**Comment by gabrieldemarmiesse** [comment]
@JoeLoser the conflicts are not easy to fix. I opened #2581 to merge as many things as possible while I work on the conflicts. The `Formatter` struct was not made to support multiple ways of formatting the same struct, we'll need some changes to this.

**Comment by gabrieldemarmiesse** [comment]
Closing as this needs to be completely reworked.

**Pull request closed without merging**